src/test/shared/credentials/loginManager.test.ts

```diff
@@ -58,6 +60,7 @@ describe('LoginManager', async () => {
 
         await loginManager.login(false, sampleCredentialsProviderId)
         assert.strictEqual(setCredentialsStub.callCount, 1, 'Expected awsContext setCredentials to be called once')
+        assert.strictEqual(recordAwsSetCredentialsSpy.calledOnce, true)
```

Comment:
These tests are written to focus on the core log in / log out functionality. I'd prefer if two additional (and similarly small) tests were written, to specifically verify the recordAwsSetCredentialsSpy behavior (one when it should be called, one when it shouldn't).

The motivations are that it only needs to be tested once, and revisiting these tests in the future, it is less clear what the test is intended to exercise. Explicit tests also have the benefit of clearly expressing that the different scenarios (in this case: one true, one false) have been covered by tests.

Reply:
added 2 new tests

Reply:
Looks good. The recordAwsSetCredentialsSpy.calledOnce checks that were added to the other tests could optionally be removed.